### DISCUSSION TITLE
Load Micronaut application-*.yml files without instantiating beans

### DISCUSSION THREAD

**Discussion by Patanouk** [micronaut-projects/micronaut-core#5975] [Q&A] [answered]
Hello,

I have a use case which falls slightly outside of Micronaut usual flow, and I'm having trouble figuring how to make it fit

I want to 

1. Load Micronaut `application-*.yml` properties files (with environment deduction : `Kubernetes`, `Test` etc...) without instantiating the beans
2. Run some code using these properties before the Micronaut application starts
3. Start the Micronaut application

Is there a way to do this in Micronaut?

----
To be more precise about my use case, I have an application which creates some kafka streams  
The kafka streams are closed automatically if the input topic doesn't exist, so I wanted to create the topics before the application starts

**Comment by jeffscottbrown** [comment, marked as answer]
> Run some code using these properties before the Micronaut application starts

If you want to load config files (or do anything else) before Micronaut gets involved, the best way to do that is to invoke that code from your `main` method before starting Micronaut.

Depending on what it is you are going to do with the config values, there may be a better thing to do that doesn't require you doing anything before engaging Micronaut.
